bgpd/bgp_zebra.c

```diff
@@ -750,20 +750,66 @@ bool bgp_zebra_nexthop_set(union sockunion *local, union sockunion *remote,
 						      peer->bgp->vrf_id);
 	}
 	if (local->sa.sa_family == AF_INET6) {
-		memcpy(&nexthop->v6_global, &local->sin6.sin6_addr, IPV6_MAX_BYTELEN);
+
+		if (!bgp_flag_check(peer->bgp,
+				BGP_FLAG_IPV6_LINK_LOCAL_ONLY))
+			memcpy(&nexthop->v6_global, &local->sin6.sin6_addr,
+				IPV6_MAX_BYTELEN);
+
+		if (BGP_DEBUG(zebra, ZEBRA)) {
+
+			char buf[INET6_ADDRSTRLEN] = {0};
+
+			zlog_debug("%s : peer %s, ipv6_local :%s",
```

Comment:
just for pedantic reasons: `zlog_debug("%s: peer %s, ipv6_local: %s", ..`

Reply:
and please remove some new empty lines

Reply:
Empty spaces removed.

Reply:
could you use please also `"%s: peer %s, ipv6_local: %s"` (notice spaces before/after `:`) Just wanna keep the code clean and properly formatted everywhere and not the way everyone likes :)

Reply:
Sure will do it.